More MKS UI cleanup, optimization

rhapsodyv opened the pull request:
### Description

Follow up for 19843 when TMC is enabled. Added a test for that too.

### Benefits

Fix #19867 

### Related Issues

#19867 
#19843

rhapsodyv committed:
follow up fixes for 19843 when TMC is enabled

rhapsodyv committed:
add new test: LVGL+TMC

rhapsodyv committed:
missing tmc util method

rhapsodyv committed:
add the big-button method

rhapsodyv commented on Marlin/src/lcd/extui/lib/mks_ui/draw_ready_print.cpp:
```diff
@@ -237,34 +237,9 @@ void lv_draw_ready_print(void) {
-    #endif
+    lv_big_button_create(scr, "F:/bmp_tool.bin", main_menu.tool, 20, 90, event_handler, ID_TOOL);
+    lv_big_button_create(scr, "F:/bmp_set.bin", main_menu.set, 180, 90, event_handler, ID_SET);
+    lv_big_button_create(scr, "F:/bmp_printing.bin", main_menu.print, 340, 90, event_handler, ID_PRINT);
```
@thinkyhead this is what I meant about the "mks button". I created it: a function that create a full mks button, with label, and everything. So, more and more code can be removed.

thinkyhead replied:
I would have done more, but I had to sleep! I'm kind of surprised no one ever optimized this code before.

rhapsodyv committed:
tool screen with big buttons

rhapsodyv committed:
set screen using big buttons

rhapsodyv committed:
home menu using big buttons

rhapsodyv committed:
move screen using big buttons

rhapsodyv committed:
extrusion using big button

rhapsodyv committed:
preheat screen using big button

rhapsodyv committed:
filament_change using big button

rhapsodyv committed:
lang screen using big button

rhapsodyv committed:
fan screen using big button

rhapsodyv committed:
helper to create lvgl menu item

rhapsodyv committed:
machine params using menu item

rhapsodyv committed:
machine settings using menu item

rhapsodyv committed:
motor settings using menu item

thinkyhead commented:
I'm pretty sure I made code that uses `toggle_stepping_mode` expect the new value, not the old value. If this is not the case, please change the code that uses `toggle_stepping_mode` instead of changing `toggle_stepping_mode`.

thinkyhead commented:
Ah, I see you are adding your own. Well, please use this instead:
```cpp
inline bool toggle_stepping_mode() { set_stealthChop(!this->stored.stealthChop_enabled); return get_stealthChop(); }
```
Yes, it does return the actual hardware status, so if toggle is failing to change the hardware state, it will be obvious in the interface.

rhapsodyv committed:
helper for menus with editable values

thinkyhead committed:
Better toggle_stepping_mode

rhapsodyv committed:
Merge branch 'followup-tmc-19843' of github.com:rhapsodyv/Marlin into followup-tmc-19843

rhapsodyv commented:
I created helpers to draw every time of "default" lvgl button and menu. I'm updating the code to use the new helpers, and removing the repeated code.
I think there're more up to 10 screens to update... this is endless...

thinkyhead committed:
Pass by reference

thinkyhead commented:
> I think there're more up to 10 screens to update... this is endless...

Welcome to my world! But the repetitive things that I can automate, I do tend to automate. For example, today I am adding the option to download and build one or more example configs to my `mftest` script as a basis for future CI on the Configurations repo. And, well, it would also produce BINs for all the configs, so those could be posted someplace too. But I digress……

thinkyhead commented on Marlin/src/lcd/extui/lib/mks_ui/draw_acceleration_settings.cpp:
```diff
@@ -160,200 +160,59 @@ static void event_handler(lv_obj_t * obj, lv_event_t event) {
 
-    buttonXValue = lv_btn_create(scr, PARA_UI_VALUE_POS_X, PARA_UI_POS_Y * 4 + PARA_UI_VALUE_V, PARA_UI_VALUE_BTN_X_SIZE, PARA_UI_VALUE_BTN_Y_SIZE, event_handler, ID_ACCE_X);
-    labelXValue = lv_label_create_empty(buttonXValue);
+    ZERO(public_buf_l);
```
I don't think these are needed in front of string population. The `sprintf_P` function is not likely to bail and return without writing to the buffer.

rhapsodyv replied:
True.. I'm too much focused in moving and rewriting those menus that something pass. I will remove all those zero.

rhapsodyv committed:
unneeded ZERO(bug)

thinkyhead committed:
Extraneous buffer zeroing

thinkyhead committed:
Merge branch 'followup-tmc-19843' of https://github.com/rhapsodyv/Marlin into pr/19869

rhapsodyv committed:
max feed rate screen using new helpers

rhapsodyv committed:
Merge branch 'followup-tmc-19843' of github.com:rhapsodyv/Marlin into followup-tmc-19843

rhapsodyv reviewed (commented):
> Ah, I see you are adding your own. Well, please use this instead:
> 
> ```c++
> inline bool toggle_stepping_mode() { set_stealthChop(!this->stored.stealthChop_enabled); return get_stealthChop(); }
> ```
> 
> Yes, it does return the actual hardware status, so if toggle is failing to change the hardware state, it will be obvious in the interface.

set_stealthChop don't exist yet..

thinkyhead committed:
Add set_stealthChop too

rhapsodyv committed:
advanced settins using draw helpers

thinkyhead committed:
Merge branch 'followup-tmc-19843' of https://github.com/rhapsodyv/Marlin into pr/19869

thinkyhead commented on Marlin/src/lcd/extui/lib/mks_ui/draw_tmc_step_mode_settings.cpp:
```diff
@@ -260,7 +217,9 @@ void lv_draw_tmc_step_mode_settings(void) {
     lv_ex_line(line1, line_points[0]);
 
-    buttonYText = lv_btn_create(scr, NULL, PARA_UI_POS_X, PARA_UI_POS_Y * 2, PARA_UI_VALUE_SIZE_X, PARA_UI_SIZE_Y, event_handler, 0);
+    buttonYText = lv_btn_create(scr, PARA_UI_POS_X, PARA_UI_POS_Y * 2, PARA_UI_VALUE_SIZE_X, PARA_UI_SIZE_Y, event_handler, 0);
```
Oops! I must have had an imgbtn fart here…

rhapsodyv committed:
more screens using draw helpers

thinkyhead committed:
Label and Button buttons

thinkyhead committed:
Merge branch 'followup-tmc-19843' of https://github.com/rhapsodyv/Marlin into pr/19869

rhapsodyv committed:
more screens using draw helpers

thinkyhead committed:
More cleanups

thinkyhead committed:
Merge branch 'followup-tmc-19843' of https://github.com/rhapsodyv/Marlin into pr/19869

thinkyhead committed:
Move extended lv declarations

thinkyhead committed:
Fix ternary typo

thinkyhead committed:
Fix style_btn_pr

rhapsodyv committed:
helper for menus with two edit fields

rhapsodyv commented:
it's never ends...

thisiskeithb commented:
Here's another bug that came out of that MKS UI slim down: https://github.com/MarlinFirmware/Marlin/issues/19872

rhapsodyv committed:
more

rhapsodyv committed:
more two

rhapsodyv committed:
more two

rhapsodyv committed:
more...

rhapsodyv committed:
helper for switch menu and one more screen

rhapsodyv committed:
the last one?!!

rhapsodyv commented:
I think I finished. I have refactored all simple/standard/template screens (menus/editors/switch/big buttons).

Russian Lang worked for me.

I tested what I could. I think everything is ok... but as I changed almost all screens, I can be missing one.

thinkyhead committed:
Apply nullptr, add lv_screen_create

thinkyhead committed:
Clean up event handling

thinkyhead committed:
Fix lv_screen_create

thinkyhead committed:
Use enum for index ids

thinkyhead committed:
Fix bugs, consolidate more

thinkyhead committed:
Add DIALOG_IS macro

thinkyhead committed:
Good enough … for now
